### ISSUE TITLE
GIFBUILDER not working anymore with DIP activated

### ISSUE THREAD

**Issue by thommyhh** [webcoast-dk/deferred-image-processing#5]
From Tyrone Rogers via TYPO3 Slack:
> Just a small note: I use Gifbuilder for watermarks etc. and this doesnt work anymore when "DIP" is activated. `
> ```php
> $conf=array(
>     'XY' => '1920,1080',
>     'format' =>'png',
>     'quality' =>'95',
>     'backColor' => '#333333',
>     '10' => 'IMAGE',
>     '10.'=> array(
>         'file' => 'fileadmin/xy.jpg',
>         'file.' => array(						
>             'width' => '1920c',
>             'height' => '1080c',
>         ),
>     ),	
>     '15' => 'IMAGE',
>     '15.'=> array(
>         'offset' => '[10.w]-270,[10.h]-235',
>         'file' => 'fileadmin/mylogo.png',
>         'file.' => array(
>             'width' => '180',
>             'height' => '180',
>         ),
>     ),
>     ...
> );
> $this->gifCreator->start($conf,$data);
> $this->gifCreator->gifBuild()
> ```
> Is there any way to make Gifbuilder wait until the images are beeing processed?

**Comment by maechler** [comment]
Maybe you could pass `'deferred' => true` to the processing instructions in order to immediately render the image.

```php
'file.' => array(
    'width' => '180',
    'height' => '180',
    'deferred' => true,
)
````

I have not tested this with the GIF creator, we did however something similar for an API endpoint where we also could not defer the rendering.

**Comment by thommyhh** [comment]
@maechler Unfortunately only certain properties of the `file.` array are used as configuration for the processing task. But I think I found another solution.

**Comment by thommyhh** [comment]
@maechler Do you use GIFBUILDER and can test it with the current master?

**Comment by maechler** [comment]
@thommyhh No, unfortunately not. I just thought the issue looked very similar to what we could solve by passing `'deferred' => true`. Maybe the reporter of the issue could test it?

**Comment by thommyhh** [comment]
@maechler Yes. I asked him, too.

**Comment by Geddo** [comment]
Hey folks, sorry to say but this didnt helped with v1.0.2

**Comment by thommyhh** [comment]
@Geddo v1.0.2 does not contain the fix for the GIFBUILDER. Try the current master or wait until tomorrow. I will release v1.0.3 tomorrow including the fix for the GIFBUILDER.

**Comment by thommyhh** [comment]
Solved with [v1.0.3](https://github.com/webcoast-dk/deferred-image-processing/releases/tag/v1.0.3).

**Comment by Geddo** [comment]
Great - works now.
